### PR TITLE
Fix handling of "off" in encryption_enabled_by_default_for_room_type

### PR DESCRIPTION
**Pull request by babolivier**
Fixes https://github.com/matrix-org/synapse/issues/7821, introduced in https://github.com/matrix-org/synapse/pull/7639

Turns out PyYAML translates `off` into a `False` boolean if it's
unquoted (see https://stackoverflow.com/questions/36463531/pyyaml-automatically-converting-certain-keys-to-boolean-values),
which seems to be a liberal interpretation of this bit of the YAML spec: https://yaml.org/spec/1.1/current.html#id864510

An alternative fix would be to implement the solution mentioned in the
SO post linked above, but I'm aware it might break existing setups
(which might use these values in the configuration file) so it's
probably better just to add an extra check for this one. We should be
aware that this is a thing for the next times we do that though.

I didn't find any other occurrence of this bug elsewhere in the
codebase.

### CODE REVIEW

File: synapse/config/room.py [hunk 1]
```diff
@@ -50,7 +50,12 @@ def read_config(self, config, **kwargs):
                 RoomCreationPreset.PRIVATE_CHAT,
                 RoomCreationPreset.TRUSTED_PRIVATE_CHAT,
             ]
-        elif encryption_for_room_type == RoomDefaultEncryptionTypes.OFF:
+        elif (
+            encryption_for_room_type == RoomDefaultEncryptionTypes.OFF
+            or not encryption_for_room_type
```
Comment (auscompgeek) [inline synapse/config/room.py:55]: Would an `is False` check be better here? Otherwise other falsy values (like None or empty string) will match.

Comment (babolivier) [reply synapse/config/room.py]: Good shout, especially as we generally try to go in that direction. Fixed in https://github.com/matrix-org/synapse/pull/7822/commits/68a25ed14c86add34ac0db46e1461b381b953a5f

**Comment by babolivier** [comment]
Yes, that's what I meant by

> I didn't find any other occurrence of this bug elsewhere in the
> codebase.

**Comment by clokep** [comment]
> Yes, that's what I meant by
> 
> > I didn't find any other occurrence of this bug elsewhere in the
> > codebase.

I guess it helps if I read the whole description. 🤦  Thanks!

File: synapse/config/room.py [hunk 2]
```diff
@@ -50,7 +50,12 @@ def read_config(self, config, **kwargs):
                 RoomCreationPreset.PRIVATE_CHAT,
                 RoomCreationPreset.TRUSTED_PRIVATE_CHAT,
             ]
-        elif encryption_for_room_type == RoomDefaultEncryptionTypes.OFF:
+        elif (
+            encryption_for_room_type == RoomDefaultEncryptionTypes.OFF
+            or encryption_for_room_type is False
+        ):
+            # PyYAML translates "off" into False if it's unquoted, so we also need to
+            # check for encryption_for_room_type being False.
```
Comment (DMRobertson) [inline synapse/config/room.py:58]: This is not a PyYAML quirk; [this is part of YAML itself](https://yaml.org/type/bool.html); [c.f. the famous Norway Problem.](https://hitchdev.com/strictyaml/why/implicit-typing-removed/)

**Pull request merged**